benfluleck/HelloBooks
# #151972556 Implement feedback for logic on borrow books controller

#### What does this PR do?
- Implements the borrow book controller with more contextualized validations
#### Description of Task to be completed?
-  Implement feedback for logic on borrow books controller
#### How should this be manually tested?
- Use the auth/user/signup route to create a new user
- Log into auth/user/signin 
- See a list of books a user can borrow with get /books
- Use:
         - POST: /users/:userId/books to borrow a book
         - GET: /users/:userId/books to view a list of borrowed books
         - PUT: /users/:userId/books to return a book

#### Any background context you want to provide?
#### What are the relevant pivotal tracker stories?
[Finishes #151972556]
